### PR TITLE
Tidy template helper methods

### PR DESCRIPTION
**Pull request by chrusty**
@deepmap-marcinr Here is what I've got so far. There is still some more stuff I'd like to improve, and some tests I'd like to run, but it should give you an idea of where I'm going with it.

### CODE REVIEW

File: pkg/codegen/operations.go [hunk 1]
```diff
@@ -294,6 +294,7 @@ func (o *OperationDefinition) GetResponseTypeDefinitions() ([]TypeDefinition, er
 
 					td := TypeDefinition{
 						TypeName: typeName,
+						JsonName: responseName,
```
Comment (chrusty) [inline pkg/codegen/operations.go]: Using this to convey the responseName. Not sure if it is better to add a new attribute to pollute `TypeDefinition`.

File: pkg/codegen/template_helpers.go [hunk 2]
```diff
@@ -130,26 +136,9 @@ func genResponseUnmarshal(operationID string, responses openapi3.Responses) stri
 		// If we made it this far then we need to handle unmarshaling for each content-type:
 		sortedContentKeys := SortedContentKeys(responseRef.Value.Content)
 		for _, contentTypeName := range sortedContentKeys {
-			contentType, ok := responseRef.Value.Content[contentTypeName]
```
Comment (chrusty) [inline pkg/codegen/template_helpers.go]: Managed to remove quite a bit of cruft

File: pkg/codegen/operations.go (hunk 1 above)
Comment (deepmap-marcinr) [reply pkg/codegen/operations.go]: I wouldn't overload the meaning of a field. Add a ResponseName and say what its purpose is. I've found that it's better to be overly verbose in this code, it's damn complex, and this kind of field separation helps inevitable future refactoring.

Comment (chrusty) [reply pkg/codegen/operations.go]: Agreed. I've added a new attribute.

**Pull request merged**